analysis/bulk-deconvolution/exploratory-notebooks/quantiseq-tumor-genes.Rmd

```diff
@@ -0,0 +1,264 @@
+  show_col_types = FALSE
+)
+map_symbols <- map_table$gene_symbol
+
+all(signature_genes %in% map_symbols)
+```
+
+We are indeed missing some! Which ones?
+
+```{r}
+signature_genes[which(!(signature_genes %in% map_symbols))]
+```
+
+It turns out the reason these are "missing" is because they have been deprecated since `quanTIseq` was written, or they are aliases for the official gene symbol:
+
+- `AKAP2` was deprecated and replaced with `PALM2AKAP2` (source: <https://www.ncbi.nlm.nih.gov/gene/11217>)
+- `FAM46C` is an alias for `TENT5C` (source: <https://www.ncbi.nlm.nih.gov/gene/54855>)
+- `GUCY1A3` is an alias for `GUCY1A1` (source: <https://www.ncbi.nlm.nih.gov/gene/2982>)
+
+Are these gene symbols present in our ensembl <-> symbol mapping table?
+
+```{r}
+updated_gene_names <- c("PALM2AKAP2", "TENT5C", "GUCY1A1")
+all(updated_gene_names %in% map_symbols)
+```
+
+Great, we have these other versions of the gene symbols!
+We'll directly replace them in the signature list here.
+
+```{r}
+signature_genes <- dplyr::case_match(
+  signature_genes, 
+  "AKAP2" ~ "PALM2AKAP2", 
+  "FAM46C" ~ "TENT5C", 
+  "GUCY1A3" ~ "GUCY1A1", 
+  .default = signature_genes
+)
+```
+
+When we actually run `quanTIseq`, we'll need to make sure that we have matching gene names between our TPM matrices and their signature.
+
+## Prepare the data
+
+We'll read in the TPM data from all projects into a single long data frame, retaining only the signature genes with an indicator column for which are the tumor genes.
+
+```{r}
+# Create data frame of TPM values for all samples
+tpm_df <- tpm_files |>
+  purrr::map(
+    \(x) {
+      # read matrix and convert to table
+      readr::read_rds(x) |>
+        as.data.frame() |>
+        tibble::rownames_to_column(var = "gene_symbol")
+      }
+  ) |> 
+  # Combine into a single data frame
+  purrr::reduce(dplyr::left_join, by = "gene_symbol") |>
+  # Filter to signature genes only and add indicator column for tumor genes
+  dplyr::filter(gene_symbol %in% signature_genes) |>
+  dplyr::mutate(tumor_gene = gene_symbol %in% tumor_genes) |>
+  # Long data frame
+  tidyr::pivot_longer(
+    starts_with("SCPCS"), 
+    names_to = "sample_id", 
+    values_to = "tpm"
+  )
+
+head(tpm_df)
+```
+
+And confirm once more that we have all the genes:
+
+```{r}
+all(signature_genes %in% tpm_df$gene_symbol)
+```
+
+## Expression distributions
+
+First, we'll visualize distributions of TPM for tumor and non-tumor genes.
+
+
+```{r fig.height=12, fig.width=16}
+ggplot(tpm_df) + 
+  aes(x = tumor_gene, y = tpm, color = tumor_gene) + 
+  geom_jitter(width = 0.1, size = 0.25) + 
+  facet_wrap(~sample_id, scales = "free_y") + 
+  theme(
+    axis.text = element_text(size = rel(0.8)), 
+    axis.title = element_text(size = rel(0.8)), 
+    legend.position = "none"
+  )
+```
+
+Looking over a great many jitter plots, there does not seem to be a universal trend of abberantly high expression among tumor genes.
+
+
+## Permutation tests
```

Comment:
Okay, but why not just do the straight tests? I would expect the non-parametric test to work just fine in this situation (and probably the t test too, since it tends to be pretty robust).

Reply:
These tests were brought to you by my rough estimation that about 10% of the meetings I go to these days feature at least a brief discussion of why more people need to be shuffling data 😂  
So, I figured, perhaps today is my turn to shuffle data!